.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,30 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/python:1-3.12-bullseye",
```

Comment:
Try `bookworm`? Also try `3.11`? I think that would match the esphome image

Reply:
I will give it a try! However on GitHub Actions (CI) it doesn't run the full Docker image, just a default runner with Python 3.12. I will see how it runs with 3.11.

Interesting that you don't face the compile issue, but I do face them on the latest ESPHome addon in HaOS and on my dev environments..